admin-ui/plugins/auth-server/components/Scopes/ScopeForm.js

```diff
@@ -31,38 +31,67 @@ import GluuTooltip from 'Routes/Apps/Gluu/GluuTooltip'
 import { SCOPE } from 'Utils/ApiResources'
 import { useTranslation } from 'react-i18next'
 import { ThemeContext } from 'Context/theme/themeContext'
-import {
-  LIMIT,
-  PATTERN,
-} from 'Plugins/auth-server/common/Constants'
+import { LIMIT, PATTERN } from 'Plugins/auth-server/common/Constants'
+import { getUMAResources } from '../../redux/actions/ScopeActions'
 
-function ScopeForm({ scope, scripts, attributes, handleSubmit}) {
+function ScopeForm({ scope, scripts, attributes, handleSubmit }) {
   const { t } = useTranslation()
   let dynamicScopeScripts = []
+  let umaResourcesScript = []
+  let umaAuthorizationPolicies = []
+
   const theme = useContext(ThemeContext)
   const selectedTheme = theme.state.theme
   const history = useHistory()
   const spontaneousClientScopes = scope.attributes.spontaneousClientScopes || []
-   const dispatch = useDispatch()
-   const client = useSelector((state) => state.oidcReducer.items)
+  const dispatch = useDispatch()
+  const client = useSelector((state) => state.oidcReducer.items)
+  const umaResources = useSelector((state) => state.scopeReducer.umaResources)
+  // const scripts = useSelector((state) => state.customScriptReducer.items)
   let claims = []
   scripts = scripts || []
   attributes = attributes || []
   dynamicScopeScripts = scripts
     .filter((item) => item.scriptType == 'DYNAMIC_SCOPE')
     .filter((item) => item.enabled)
     .map((item) => ({ dn: item.dn, name: item.name }))
+
+  umaAuthorizationPolicies = scripts
+    .filter((item) => item.scriptType == 'UMA_RPT_POLICY')
+    .filter((item) => item.enabled)
+    .map((item) => ({ dn: item.dn, name: item.name }))
+
+  umaResourcesScript = umaResources.map((item) => ({
+    dn: item.dn,
+    name: item.id,
+  }))
+
+  useEffect(() => {
+    dispatch(getUMAResources({ pattern: 'scope' }))
+  }, [])
+
   claims = attributes.map((item) => ({ dn: item.dn, name: item.displayName }))
 
   const [init, setInit] = useState(false)
   const [modal, setModal] = useState(false)
-  const [showClaimsPanel, handleClaimsPanel] = useState(scope.scopeType === 'openid')
-  const [showDynamicPanel, handleDynamicPanel] = useState(scope.scopeType === 'dynamic')
-  const [showSpontaneousPanel, handleShowSpontaneousPanel] = useState(scope.scopeType === 'spontaneous')
+  const [showClaimsPanel, handleClaimsPanel] = useState(
+    scope.scopeType === 'openid',
+  )
+  const [showDynamicPanel, handleDynamicPanel] = useState(
+    scope.scopeType === 'dynamic',
+  )
+  const [showSpontaneousPanel, handleShowSpontaneousPanel] = useState(
+    scope.scopeType === 'spontaneous',
+  )
+  const [showUmaPanel, handleShowUmaPanel] = useState(scope.scopeType === 'UMA')
 
   useEffect(() => {
     if (showSpontaneousPanel) {
-      dispatch(searchClients({ "action_data": makeOptions(1, scope.attributes.spontaneousClientId) }))
+      dispatch(
```

Comment:
This is not required. Please populated associated clients for UMA and spontaneous scopes from scope obj.

Reply:
removed